File: go/27.remove-element.go
/*
 * @lc app=leetcode id=27 lang=golang
 *
 * [27] Remove Element
 */
func removeElement(nums []int, val int) int {
	if len(nums) == 0 {
		return 0
	}
	
	for i := 0; i < len(nums) - 1; {
		v := nums[i]
		if v == val {
			nums = append(nums[:i], nums[i+1:]...)
		} else {
			i++
		}
	}

	if nums[len(nums) - 1] == val {
		nums = nums[:len(nums) - 1]
	}

	return len(nums)
}

File: csharp/27.remove-element.cs
/*
 * @lc app=leetcode id=27 lang=csharp
 *
 * [27] Remove Element
 */

// @lc code=start
public class Solution 
{
    public int RemoveElement(int[] nums, int val) 
    {
        int left = 0;
        int right = nums.Length - 1;

        while (left <= right)
        {
            if (nums[left] == val)
            {
                if (nums[right] == val)
                {
                    right--;
                }
                else 
                {
                    nums[left] = nums[right];
                    left++;
                    right--;
                }
            }
            else 
            {
                left++;
            }
        }

        return right + 1;
    }
}
// @lc code=end

